.github/pull_request_template.md

```diff
@@ -6,12 +6,13 @@ Please write a few sentences describing the overall goals of the pull request's
 
 ## PR checklist
 
-Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")
+Please add/remove strikethrough ~~ as appropriate, and add any relevant link/s to the end of the line.
 
-- [ ] **changelog** provided, or not required
-- [ ] **3.6 backport** done, or not required
-- [ ] **2.28 backport** done, or not required
-- [ ] **tests** provided, or not required
+- [ ] **changelog** provided, ~~not required~~, ~~covered~~ . Link --> #XYZ
+- [ ] **dev PR** provided, ~~not required~~ Link --> #XYZ
+- [ ] **3.6 PR** provided, ~~not required~~ Link --> #XYZ
+- [ ] **2.28** PR provided, ~~not required~~ Link --> #XYZ
+- **tests**  provided, ~~not required~~, ~~covered~~ Link --> #XYZ
```

Comment:
To be honest I am a bit torn with Tests and changelog. Yes you are right with the idea of that they should be self contained, and either present or explicitely stated why they are not.

And unlike the linking with backports being on the PR template does not offer much more information than if we used two simple labels such as `no-changelog`, `no-tests` to indicate the use-cases that are not required.

I am happy to go whichever direction the team wants.

Reply:
In an ideal world, we would always add complete testing with bug fixes and new features. And I agree that we shouldn't encourage adding less testing. However, in practice we have exceptions more often than we would like to: bug fixes where adding tests is prohibitively expensive, new feature development where only limited testing can be added at that point (eg. unit tests or compatibility tests, simply because we don't have enough of the feature implemented to do more, or for practical reasons where we can't keep the PRs small and reviewable in any other way than only adding smoke tests and extensive testing comes in a separate PR) I thought that this line could be useful for providing justification in these cases. As a reviewer or a gatekeeper I would prefer to see some justification as opposed to an empty ~tests` section in the template.